### ISSUE TITLE
Setting rmax before rmin

### ISSUE THREAD

**Issue by kslong** [sirocco-rt/sirocco#427]
There are several XXX issues having to do with rmax.  In the current, dev branch (and historically), we have read in Wind.radmax --> zdom[ndom].rmax before reading in the  various parameters (including rmin for a wind model).   This seemed quite illogical, so ksl tried to move this question until after we had read in the other parameters.  This however caused several wind models to fail, most notably homologous (for which there is a separate XXX issue about this), and Nick's hydro models.  

Further investigation of the homologous case shows that what fails is not the setting up of the grid, but the determination of which cells are in the wind, and the reason this fails is that within the routine which sets up a homologous domain,get_homologous_params, one sets up zdom[ndom].rhomax and zdom[ndom].zmax.  Since these are not correct, the routines that determine which cells are in the grid are wrong ... that is none of the cells appear to be in the wind.

A similar situation undoubtedly appears in the hydro case.

There are two solutions to this

- We add parameters within get_homologous_params and get_hydro_wind_params (and any other model that we have missed this) to get zdom[ndom].rmax.   This has the advantage that the parameter names will be clearer, and that everything will be self contained.  It also means things can be tailored to the individual model.  But some parameter names will change (which in principle can be dealt with synonyms)
- We pull establishing the outside edges of the a wind component outside of the various get_params routines.

ksl is in favor of option 1, and plans to implement this, unless there are strong objections from @Higginbottom   Note, for hydro we can in principle get this from the grid itself.

**Comment by kslong** [comment]
Note - weirdly homologous used to have radmax read in, and the .pf in examples includes this.  Someone probably, ksl, must have changed this in a possibly misguided attempt to make things more uniform.  

There are some other variables like initial temperature and clumping that are similar in the sense that every wind component has to have a temperature and a clumping factor.  and these are set outside of the individual get_param routines in setup_domains.  We proably do not want to pull these into the individual get_param routines.  We always need a radmax, but it's obviously tightly coupled to the geometry.

Note - Looking at some of the other get_params routines, notably sv, the logic for establish the maximum radius, etc is a bit obscure.

Note - need to be careful with synonyms.  They can map many to one, but not one to many.  If keywords are added like homologous.radmax and hydro.radmax, and these replace wind.radmax this should be OK.

**Comment by kslong** [comment]
What I have decided to do for now is, to add a parameter where it is required to know zoom[].rmax within the routine, e.g in the shell model and in the stellar_wind model.  There are a number of other models that do not require  zoom[ndom].rmax within the get_params routines and I have left these alone.  For these, one is now asked for wind_rmax after the other parameters are obtained, instead of before any of them as previously.  The program issues an error in these cases just to remind me that everything is not totally self-consistent.  

For the yso model, I have tested with the same model James used for #250.  I have placed the field for that here for consistency

[yso_a.pf.txt](https://github.com/agnwinds/python/files/2311951/yso_a.pf.txt)
[yso_ionization.dat.txt](https://github.com/agnwinds/python/files/2311953/yso_ionization.dat.txt)

Aside: It is unclear whether we should retain the yso model at all.  It should be implemented with domains, and so might be deleted altogether.

At present I am still testing these changes.
